pci: report NUMA node locality for the PCI devices

fromanirh opened the pull request:
We want to expose the NUMA locality of the PCI devices,
much like we already do for GPU devices.
This is very important for high-end, high-performance devices
(NICs, storage controllers).

To enable testing, we update the snapshot with another one
captured with more PCI device informations.
See: https://github.com/jaypipes/ghw/pull/224

Signed-off-by: Francesco Romani <fromani@redhat.com>

fromanirh commented:
(Implicitely) depends on https://github.com/jaypipes/ghw/pull/224

jaypipes commented on pkg/topology/topology.go:
```diff
@@ -79,7 +79,13 @@ type Info struct {
+// NewWithContext returns a pointer to an Info struct that contains information about
+// the NUMA topology on the host system. Use this function when you want to consume
+// the topology package from another package (e.g. pci, gpu)
+func NewWithContext(ctx *context.Context) (*Info, error) {
```
:+1:

jaypipes reviewed (approved):
@fromanirh I'm happy with this, all looks good to me. Are we good to merge this one?

fromanirh commented:
hi Jay! We can merge this PR (pending me addressing your comments :) ) with the caveat that `ghw-snapshot` cannot create valid snapshot to exercise the code added here (e.g. in the tests) until https://github.com/jaypipes/ghw/pull/224 is merged

fromanirh commented on pkg/pci/pci.go:
```diff
@@ -39,6 +40,9 @@ type Device struct {
 	Subclass *pcidb.Subclass `json:"subclass"`
 	// optional programming interface
 	ProgrammingInterface *pcidb.ProgrammingInterface `json:"programming_interface"`
+	// Topology node that the graphics card is affined to. Will be nil if the
```
yeah, copy/paste strikes again :)

fromanirh commented:
And btw, after some trial/errors, https://github.com/jaypipes/ghw/pull/224 reached a form I'm comfortable with, and requires just a minor followup now.

jaypipes reviewed (approved):
:+1:  nice work Francesco
